src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/TransitionalAccommodationReferralReportRepository.kt

```diff
@@ -0,0 +1,74 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.repository
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingEntity
+import java.time.LocalDate
+import java.util.UUID
+
+interface TransitionalAccommodationReferralReportRepository : JpaRepository<BookingEntity, UUID> {
+  @Query(
+    """
+    SELECT cast(a.id as text) AS assessmentId,
+      CAST(ap.id as text) AS referralId,
+      ap.created_at AS referralCreatedDate,
+      ap.crn AS crn,
+      ap.submitted_at AS referralSubmittedDate,
+      taa.risk_ratings->'roshRisks'->'value'->>'overallRisk' AS riskOfSeriousHarm,
+      taa.is_registered_sex_offender AS sexOffender,
+      taa.needs_accessible_property AS needForAccessibleProperty,
+      taa.has_history_of_arson AS historyOfArsonOffence,
+      taa.is_duty_to_refer_submitted AS dutyToReferMade,
+      taa.duty_to_refer_submission_date AS dateDutyToReferMade,
+      taa.duty_to_refer_local_authority_area_name AS dutyToReferLocalAuthorityAreaName,
+      probation_region.name AS probationRegionName,
+      a.decision AS assessmentDecision,
+      a.rejection_rationale AS assessmentRejectionReason,
+      a.submitted_at AS assessmentSubmittedDate,
+      CAST(b.id AS text) AS bookingId,
+      taa.is_eligible AS isReferralEligibleForCas3,
+      taa.eligibility_reason AS referralEligibilityReason,
+      ap.noms_number as nomsNumber
+    FROM temporary_accommodation_assessments aa
+    JOIN assessments a on aa.assessment_id = a.id AND a.service='temporary-accommodation' AND a.reallocated_at IS NULL
+    JOIN applications ap on a.application_id = ap.id AND ap.service='temporary-accommodation'
+    LEFT OUTER JOIN temporary_accommodation_applications taa on ap.id = taa.id
+    LEFT JOIN probation_regions probation_region ON probation_region.id = taa.probation_region_id
+    LEFT JOIN bookings b on b.application_id = ap.id AND b.service='temporary-accommodation'
+    WHERE
+      a.service = 'temporary-accommodation'
+      AND ap.submitted_at BETWEEN :startDate AND :endDate
```

Comment:
Good one, but i thought about it and decided to use assessments table because Applications can be created without completing it, but we are interested in only the completed applications. Hence i used the assessment, i will double check with BA

Reply:
cool. Just to make sure we're on the same page - this is filtering on the application's submitted_at date instead of the assessment's submitted_at date, that was my query (I agree doing a SELECT ... FROM assessments is good)

Reply:
We are using application.submitted_date because the application.submitted_date is updated with value when the assessment is successfully created. 
If there is no assessment created, then the application.submitted-date is null.